### PR TITLE
feat(backend): liquidity resolver for asset and peer

### LINKED ISSUE #1436: Allow querying asset and peer liquidity
- Add GraphQL Admin API query to display peer and asset liquidity

### PR DESCRIPTION
**Pull request by sabineschaller**
## Changes proposed in this pull request

- 

## Context

- fixes #1436 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

**Review by mkurapov — COMMENTED** [review]
I think I would prefer two separate APIs: `getPeerLiqudity` `getAssetLiquidity` since it's more explicit for the caller (since we have other `LiquidityAccountType`s)

**Review by mkurapov — COMMENTED** [review]
I think I would prefer two separate APIs: `getPeerLiqudity` `getAssetLiquidity` since it's more explicit for the caller (since we have other `LiquidityAccountType`s)

**Comment by mkurapov** [comment]
@sabineschaller 
> I wonder whether we should add liquidity to the asset and peer queries, too, in addition to having this query (or even instead of it).

I would prefer this actually, instead of this query. It would make it easier for the API caller, and would make it quite easy to update in the rafiki admin as well

**Comment by BlairCurrey** [comment]
> I wonder whether we should add liquidity to the asset and peer queries, too, in addition to having this query (or even instead of it).

I agree with Max on putting it on the asset/peer instead. 

Mostly because we are getting it by the asset or peer id already. This would prevent us from having to make 2 requests if we wanted peer liquidity and other peer info (`name`, etc.) And of course if you only wanted liquidity you could just specify that field alone. Also, I am looking at the schema and think the id in `liquidity(id: String!)` and the returned `Liquidity` type are a little unintuitive (id is not of liquidity, but asset or peer).

File: packages/backend/src/graphql/resolvers/peer.ts [hunk 1]
```diff
@@ -29,10 +29,15 @@ export const getPeers: QueryResolvers<ApolloContext>['peers'] = async (
   )
   return {
     pageInfo,
-    edges: peers.map((peer: Peer) => ({
-      cursor: peer.id,
-      node: peerToGraphql(peer)
-    }))
+    edges: await Promise.all(
+      peers.map(async (peer: Peer) => {
+        const balance = await getBalance(ctx, peer.id)
```
Comment (BlairCurrey) [inline packages/backend/src/graphql/resolvers/peer.ts:34]: This `getBalance` db call could happen up to 100 times (max page size), even when liquidity is not specified in the gql requested (making them all unnecessary). One option is to check info with [graphql-parse-resolve-info](https://www.npmjs.com/package/graphql-parse-resolve-info) first.

Another option is to use a custom resolver for liquidity, which I think solves another issue as well. I elaborate on that here https://github.com/interledger/rafiki/pull/1466/files#r1203146434

File: packages/backend/src/graphql/resolvers/asset.ts [hunk 2]
```diff
@@ -132,10 +139,26 @@ export const updateAssetWithdrawalThreshold: MutationResolvers<ApolloContext>['u
...
+  asset: Asset,
+  balance?: bigint
+): SchemaAsset => ({
   id: asset.id,
   code: asset.code,
   scale: asset.scale,
+  liquidity: balance,
   withdrawalThreshold: asset.withdrawalThreshold,
   createdAt: new Date(+asset.createdAt).toISOString()
 })
```
[the remark this answers is not in the document]
Comment (sabineschaller) [reply packages/backend/src/graphql/resolvers/asset.ts:164]: I like that. Thank you!

**Review by BlairCurrey — APPROVED** [review]

**Pull request merged**